examples/apps/ai-collab/src/components/UserPresenceGroup.tsx

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+"use client";
+
+import { Avatar, Badge, styled } from "@mui/material";
+import React, { useEffect, useRef, useState } from "react";
+import { v4 as uuid } from "uuid";
+
+import type { UserPresence } from "@/app/presence";
+import { getProfilePhoto } from "@/infra/authHelper";
+
+interface UserPresenceProps {
+	userPresenceGroup: UserPresence;
+}
+
+const UserPresenceGroup: React.FC<UserPresenceProps> = ({
+	userPresenceGroup,
+}): JSX.Element => {
+	const photoUrlsMap = new Map<string, string>();
+	const isFirstRender = useRef(true);
+	const [photoUrls, setPhotoUrls] = useState<string[]>([]);
+	const currentUserId: `id-${string}` = `id-${uuid()}`;
+
+	/**
+	 * fetch the user's photo if it's spe client, for the tinylicious client, it will use the default photo.
+	 * */
+	const updateUserPresenceGroup = async (): Promise<void> => {
+		const clientId = process.env.NEXT_PUBLIC_SPE_CLIENT_ID;
+		const tenantId = process.env.NEXT_PUBLIC_SPE_ENTRA_TENANT_ID;
+		let photoUrl: string = "";
+
+		// spe client
+		if (tenantId !== undefined && clientId !== undefined) {
+			photoUrl = await getProfilePhoto();
+		}
+		userPresenceGroup.props.onlineUsers.local.set(currentUserId, {
+			value: { photo: photoUrl },
+		});
+
+		isFirstRender.current = false;
+	};
+
+	useEffect((): void => {
+		if (isFirstRender.current) {
+			updateUserPresenceGroup().catch((error) => console.error(error));
+		}
+
+		userPresenceGroup.props.onlineUsers.events.on("itemUpdated", (update) => {
+			photoUrlsMap.set(update.key, update.value.value.photo);
+			setPhotoUrls([...photoUrlsMap.values()]);
+			console.log(photoUrls);
+		});
+		userPresenceGroup.props.onlineUsers.events.on("itemRemoved", (update) => {
+			photoUrlsMap.delete(update.key);
+			setPhotoUrls([...photoUrlsMap.values()]);
+		});
```

Comment:
In addition to `itemRemoved`, you might want to watch for disconnected session client. If their connection is lost, then they won't be able to remove the item. You can listen to `attendeeDisconnected` (and `attendeeJoined`) on `IPresence.events` to see if someone drops (and rejoins). `attendeeDisconnected` will provide `ISessionClient` that you can delete from map. (Or delete `ISessionClient.sessionId` if you want to stick with a string.) (For join you'd need to `try` to get clientValue. Something like:
```typescript
try {
const photo = userPresenceGroup.props.onlineUsers.clientValue(client).get("photo").photo;
if (photo) { photoUrlsMap.set(client, photo.photo); }
}
catch (_e) { /* no photo yet provided - okay */ }
```
)

OR whenever you see itemUpdated/itemRemoved/attendeeJoined/attendeeDisconnected you use
```typescript
const photoUrls: string[] = [];
for (const clientInfo of userPresenceGroup.props.onlineUsers.clientValues()) {
   // Use only connected clients
   if (clientInfo.client.getConnectionStatus() === "Connected") {
      const photoUrl = clientInfo.items.get("photo")?.photo;
      if (photoUrl) { photoUrls.push(photoUrl); }
   }
}
```

Reply:
Gotcha. Updated to use attendeeJoined/attendeeDisconnected instead